packages/stryker-mocha-framework/test/integration/nestedSuiteSpec.ts

```diff
@@ -0,0 +1,81 @@
+import * as path from 'path';
+import * as execa from 'execa';
+import { expect } from 'chai';
+import { TestSelection } from 'stryker-api/test_framework';
+import MochaTestFramework from '../../src/MochaTestFramework';
+import * as fs from 'fs';
+import * as rimraf from 'rimraf';
+
+interface MochaTestRunResult {
+  tests: MochaTest[];
+  pending: MochaTest[];
+  failure: MochaTest[];
+  passes: MochaTest[];
+}
+
+interface MochaTest {
+  title: string;
+  fullTitle: string;
+}
+
+// See https://github.com/stryker-mutator/stryker/issues/249
+describe('Selecting tests with nested suites', function () {
+
+  this.timeout(10000);
+  let sut: MochaTestFramework;
+  const nestedSuiteFile = path.resolve(__dirname, '..', '..', 'testResources', 'nested-suite.js');
+  const selectTestFile = path.join(__dirname, '..', '..', 'testResources', '__filterSpecs.js');
+  const testSelections: ReadonlyArray<Readonly<TestSelection>> = [
+    { id: 0, name: 'outer test 1' },
+    { id: 1, name: 'outer test 2' },
+    { id: 2, name: 'outer inner test 3' }
+  ];
+
+  beforeEach(() => {
+    sut = new MochaTestFramework();
+  });
+
+  afterEach(() => {
+    rimraf.sync(selectTestFile);
+  });
+
+  it('should run all tests in expected order when running all tests', () => {
+    const result = execMocha(nestedSuiteFile);
+    expect(result.tests.map(test => test.fullTitle)).deep.eq(['outer test 1', 'outer test 2', 'outer inner test 3']);
+  });
+
+  it('should only run test 1 if filtered on index 0', () => {
+    filter([0]);
+    const result = execMocha(selectTestFile, nestedSuiteFile);
+    expect(result.tests).lengthOf(1);
+    expect(result.passes).lengthOf(1);
+    expect(result.passes[0].fullTitle).eq('outer test 1');
+  });
+
+  it('should only run test 2 if filtered on index 1', () => {
+    filter([1]);
+    const result = execMocha(selectTestFile, nestedSuiteFile);
+    expect(result.tests).lengthOf(1);
+    expect(result.passes).lengthOf(1);
+    expect(result.passes[0].fullTitle).eq('outer test 2');
+  });
+
+  it('should only run test 3 if filtered on index 2', () => {
+    filter([2]);
+    const result = execMocha(selectTestFile, nestedSuiteFile);
+    expect(result.tests).lengthOf(1);
+    expect(result.passes).lengthOf(1);
+    expect(result.passes[0].fullTitle).eq('outer inner test 3');
+  });
+
+  function filter(testIds: number[]) {
```

Comment:
The same filter comment for the Jasmine test applies here

Reply:
Good point. I've added a test where we filter 2 tests just to make sure it works.